tests/meta/demultiplex/test_utils.py

```diff
@@ -331,6 +335,85 @@ def test_is_syncing_complete_false(
     assert not is_directory_synced
 
 
+@pytest.mark.parametrize(
+    "source_files,expected_result",
+    [
+        ([DemultiplexingDirsAndFiles.COPY_COMPLETE], True),
+        ([DemultiplexingDirsAndFiles.OUTPUT_FILE_MANIFEST], False),
+        (
+            [
+                DemultiplexingDirsAndFiles.OUTPUT_FILE_MANIFEST,
+                DemultiplexingDirsAndFiles.COPY_COMPLETE,
+            ],
+            False,
+        ),
+        ([DemultiplexingDirsAndFiles.CUSTOM_OUTPUT_FILE_MANIFEST], False),
+        (
+            [
+                DemultiplexingDirsAndFiles.CUSTOM_OUTPUT_FILE_MANIFEST,
+                DemultiplexingDirsAndFiles.COPY_COMPLETE,
+            ],
+            False,
+        ),
+    ],
+)
+def test_needs_manifest_file(source_files: list[str], expected_result: bool, tmp_path):
+    """Tests if a manifest file is needed given the files present."""
+    # GIVEN a source directory
+    source_directory = Path(tmp_path, "source")
+    Path(tmp_path, "source").mkdir()
+
+    # GIVEN the files present
+    for file in source_files:
+        Path(source_directory, file).touch()
+
+    # WHEN checking if a manifest file is needed
+    result = needs_manifest_file(source_directory)
+
+    # THEN the result should as expected
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "source_files,expected_result",
+    [([DemultiplexingDirsAndFiles.COPY_COMPLETE], True), ([], False)],
+)
+def test_flow_cell_sync_confirmed(source_files: list[str], expected_result: bool, tmp_path):
+    """Tests the check that a flow cell sync has been confirmed."""
+    # GIVEN a flow cell directory with the given file present
+    for file in source_files:
+        Path(tmp_path, file).touch()
+
+    # WHEN checking if the flow cell sync has been confirmed
+    result = flow_cell_sync_confirmed(tmp_path)
+
+    # THEN the result should match what we expect
+    assert result == expected_result
```

Comment:
Not sure if I went overboard with the whole parametrize thing here. It might be clearer as two separate tests. What do you think?

Reply:
I think is fine.